docs/specifications/tests/DNSSEC-TP/dnssec02.md

```diff
@@ -3,125 +3,202 @@
+
+The value in the Level column is the default severity level of the message. The
+severity level can be changed in the [Zonemaster-Engine profile]. Also see the
+[Severity Level Definitions] document.
+
+The argument names in the Arguments column lists the arguments used in the
+message. The argument names are defined in the [argument list].
+
+
+## Test procedure
+
+
+
+1.  Create the following empty sets:
+    1. DS record RDATA ("DS Record").
+    2. Name server IP and key tag from DS record ("No DNSKEY for DS").
+    3. Name server IP and key tag from DS record ("No match DS DNSKEY").
+    4. Name server IP and DNSKEY record key tag ("DNSKEY not for zone signing").
+    5. Name server IP and DNSKEY record key tag ("DNSKEY not SEP").
+    6. Name server IP and DNSKEY record key tag ("No matching DNSKEY RRSIG").
+    7.  Name server IP address, DNSKEY record key tag and DNSKEY algorithm code
+        ("Algo Not Supported By ZM").
+    8. Name server IP and key tag from RRSIG record ("RRSIG not valid by DNSKEY").
+
+2.  If the *Test Type* is "[undelegated]" do:
+    1. If *Undelegated DS* is empty then do terminate this test case.
+    2. Else add *Undelegated DS* as DS records to the *DS Record* set.
+
+3.  If *Test Type* is "normal", then:
+    1. Create a DS query with the DO flag set for the name of the *Child Zone*
+       ("DS Query").
+    2. Retrieve all name server IP addresses for the parent zone of
+       *Child Zone* using [Method1] ("Parent NS IP").
+    3. For each parent name server in *Parent NS IP* do:
+       1. Send *DS Query* to the name server IP.
+       2. If at least one of the following criteria is met, then go to next
+          parent name server:
+          1. There is no DNS response.
+          2. The RCODE of response is not "NoError" ([IANA RCODE List]).
+          3. The AA flag is not set in the response.
+          4. There is no DS record with matching owner name in the answer
+             section.
+       3. Retrieve the DS records from the response and add them to the
+          *DS Record* set.
+    4. If the *DS Record* set is empty exit this test case.
+4.  Create a DNSKEY query for the *Child Zone* with the DO flag set
+    ("DNSKEY Query").
+
+5.  Obtain the set of child name server IP addresses using [Method4] and
+    [Method5] ("Child NS IP").
+
+6.  For each child name server in *Child NS IP* do:
+    1. Send *DNSKEY Query* over UDP to the name server IP and collect the response.
+    2. If at least one of the following criteria is met, then go to next
+          child name server:
+       1. There is no DNS response.
+       2. The RCODE of the response is not "NoError" ([IANA RCODE List]).
+       3. The AA flag is not set in the response.
+       4. There is no DNSKEY record with owner name matching the query in the
+          answer section.
+    3. Extract the DNSKEY RRset ("DNSKEY RRs").
```

Comment:
The same action is taken here as in step 3.3.3, but the wording is very different ("retrieve" vs. "extract"). Do you think it would be possible to make the wordings more consistent? If you're searching for something in the test case it's easier if the same words are always used for the same thing.

Reply:
Changed to "retrieved" which is probably best for the context.